### DISCUSSION TITLE
Algolia Instant search default order by state (`live`, 'published', 'draft')

### DISCUSSION THREAD

**Discussion by Montana** [algolia/instantsearch#5119] [General]
I'm trying to fetch Algolia results in this particular order, hits have `state` column, and `state` values can be `live/published/draft`:

**Data in algolia**:

```sql
{ name: Post1, state: draft},
{ name: Post2, state: published},
{ name: Post3, state: live}
```

**Expected Result**:

Default order should be, first live then published and then show draft:

```sql
{ name: Post3, state: live},
{ name: Post2, state: published},
{ name: Post1, state: draft}
```

I'm aware algolia doesn't support sorting types like this but is it feasible that I could simulate the enumerated sort by performing a multi index query with three separate queries to the same index with each of the state values as a filter, then display the results for each state?

**Comment by Haroenv** [comment]
This is more a question for ranking than for InstantSearch, but you can have a numeric state `0 (draft), 1 (published), 2 (live)` and use that as a custom ranking
